Forge.TreeWalker.UnitTests/test/TreeSchemaValidatorTests.cs

```diff
@@ -0,0 +1,166 @@
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaAsString_WithForgeTree()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(forgeTreeAsString, stringRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaAsString_WithValidateAsDictionary()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(forgeTreeDictionaryAsString, rulesForDictionary, true, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaFromPath_WithoutValidateAsDictionary()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaFromPath(forgeTreeFromPath, stringRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemasFromDirectory_WithValidateAsSeparateFiles()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaFromDirectory(schemaDirectoryPath, stringRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_GetLinkedJSchemaRules_WithInvalidUrl__Fail()
+        {
+            Assert.ThrowsException<UriFormatException>(() => ForgeSchemaValidator.GetLinkedJSchemaRules(stringRules, stringRules, "//"));
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaAsForgeTree_WithEmptyStringRules__Fail()
+        {
+            Assert.ThrowsException<ArgumentException>(() => ForgeSchemaValidator.ValidateSchemaAsForgeTree(treeSchema, "", out IList<ValidationError> errorList));
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaAsString_WithErrorContent_Fail()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(invalidSchemaWithErrorContent, jschemaRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(false, res);
+            Assert.AreEqual("JSON is valid against no schemas from 'oneOf'.", errorList.First().Message);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchema_FromInvalidDirectoryPath_Fail()
+        {
+            Assert.ThrowsException<IOException>(() => ForgeSchemaValidator.ValidateSchemaFromDirectory(invalidSchemaDirectoryPath, jschemaRules, false, out IList<ValidationError> errorList));
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaFromDirectory_WithValidateAsDictionary_DirectoryContainsForgeTrees_Fail()
```

Comment:
>Test_ValidateSchemaFromDirectory_WithValidateAsDictionary_DirectoryContainsForgeTrees_Fail [](http://example.com/codeflow?start=20&length=90)

Slight name update, since "ForgeTrees" sounds like a dictionary. We actually are trying to say there are ForgeTree objects.

DirectoryContainsForgeTree_Fail